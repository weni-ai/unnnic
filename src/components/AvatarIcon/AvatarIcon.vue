<template>
  <div :class="['unnnic-avatar-icon', scheme, size, { disabled: !enabled }]">
    <unnnic-icon :icon="icon" :scheme="enabled ? scheme : 'neutral-cloudy'" :size="iconSize" />
  </div>
</template>

<script>
import UnnnicIcon from '../Icon.vue';

export default {
  components: {
    UnnnicIcon,
  },

  props: {
    enabled: {
      type: Boolean,
      default: true,
    },

    icon: {
      type: String,
      default: 'graph-stats-1',
    },

    size: {
      type: String,
      default: 'sm',
      validator(value) {
        return ['nano', 'xs', 'sm', 'lg', 'xl'].indexOf(value) !== -1;
      },
    },

    scheme: {
      type: String,
      default: 'aux-blue',
    },
  },

  computed: {
    iconSize() {
      return {
        nano: 'xs',
        xs: 'sm',
        sm: 'md',
        lg: 'lg',
        xl: 'xl',
      }[this.size];
    },
  },
};
</script>

<style lang="scss" scoped>
@import '../../assets/scss/unnnic.scss';

$scheme-colors:
  'feedback-red' $unnnic-color-feedback-red,
  'feedback-green' $unnnic-color-feedback-green,
  'feedback-yellow' $unnnic-color-feedback-yellow,
  'feedback-blue' $unnnic-color-feedback-blue,
  'feedback-grey' $unnnic-color-feedback-grey,
   'aux-blue' $unnnic-color-aux-blue,
   'aux-purple' $unnnic-color-aux-purple,
   'aux-orange' $unnnic-color-aux-orange,
   'aux-lemon' $unnnic-color-aux-lemon,
   'aux-pink' $unnnic-color-aux-pink;

@each $name, $color in $scheme-colors {
  .unnnic-avatar-icon {
    &.#{$name} {
      background-color: rgba($color, $unnnic-opacity-level-extra-light);
    }
  }
}

.unnnic-avatar-icon.disabled {
  background-color: $unnnic-color-background-grass;
}

.unnnic-avatar-icon {
  border-radius: $unnnic-border-radius-sm;
  padding: $unnnic-spacing-inset-nano;
  display: inline-flex;

  &.nano, &.xs {
    padding: $unnnic-spacing-stack-nano $unnnic-spacing-inline-nano;
  }
}
</style>
