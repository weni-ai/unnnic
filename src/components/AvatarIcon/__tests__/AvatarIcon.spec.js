import { mount } from '@vue/test-utils';
import { describe, test, expect, beforeEach } from 'vitest';
import AvatarIcon from '@/components/AvatarIcon/AvatarIcon.vue';

describe('AvatarIcon.vue', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = mount(AvatarIcon, {
      props: {
        enabled: true,
        icon: 'graph-stats-1',
        size: 'sm',
        scheme: 'aux-blue',
        filled: false,
        opacity: true,
      },
      global: {
        stubs: {
          UnnnicIcon: true,
        },
      },
    });
  });

  test('renders the component correctly', () => {
    expect(wrapper.exists()).toBe(true);
  });

  test('applies the correct classes based on props', async () => {
    expect(wrapper.classes()).toContain('unnnic-avatar-icon');
    expect(wrapper.classes()).toContain('aux-blue');
    expect(wrapper.classes()).toContain('sm');
    expect(wrapper.classes()).not.toContain('disabled');
    expect(wrapper.classes()).not.toContain('filled');
    expect(wrapper.classes()).not.toContain('opacity');

    await wrapper.setProps({
      enabled: false,
      size: 'lg',
      filled: true,
      opacity: false,
    });
    expect(wrapper.classes()).toContain('lg');
    expect(wrapper.classes()).toContain('disabled');
    expect(wrapper.classes()).toContain('opacity');
  });

  test('renders UnnnicIcon with correct props', () => {
    const icon = wrapper.findComponent({ name: 'UnnnicIcon' });
    expect(icon.exists()).toBe(true);
    expect(icon.props('icon')).toBe('graph-stats-1');
    expect(icon.props('scheme')).toBe('aux-blue');
    expect(icon.props('size')).toBe('md');
    expect(icon.props('filled')).toBe(false);
  });

  test('computes iconSize correctly based on size prop', async () => {
    expect(wrapper.vm.iconSize).toBe('md');

    await wrapper.setProps({ size: 'nano' });
    expect(wrapper.vm.iconSize).toBe('xs');

    await wrapper.setProps({ size: 'xl' });
    expect(wrapper.vm.iconSize).toBe('xl');
  });

  test('renders neutral-cloudy scheme when enabled is false', async () => {
    await wrapper.setProps({ enabled: false });
    const icon = wrapper.findComponent({ name: 'UnnnicIcon' });
    expect(icon.props('scheme')).toBe('neutral-cloudy');
  });

  test('applies opacity class based on prop', async () => {
    console.log('wrapper', wrapper.classes(), wrapper.vm.opacity);
    expect(wrapper.classes()).not.toContain('opacity');

    await wrapper.setProps({ opacity: false });
    expect(wrapper.classes()).toContain('opacity');
  });

  test('matches the snapshot', () => {
    expect(wrapper.html()).toMatchSnapshot();
  });
});
