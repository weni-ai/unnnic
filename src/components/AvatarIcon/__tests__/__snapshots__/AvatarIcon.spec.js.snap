// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`AvatarIcon.vue > matches the snapshot 1`] = `
"<div data-v-5f185a90="" class="unnnic-avatar-icon aux-blue sm">
  <unnnic-icon-stub data-v-5f185a90="" filled="false" icon="graph-stats-1" clickable="false" size="md" scheme="aux-blue"></unnnic-icon-stub>
</div>"
`;
